runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -258,22 +259,23 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	if (NULL != spine) {
 		switch (_layout) {
 		case GC_ArrayletObjectModel::Discontiguous:
-			/* if last arraylet leaf is empty (contains 0 bytes) arrayoid pointer is set to NULL */
-			if (arrayoidIndex == (_numberOfArraylets - 1)) {
-				Assert_MM_true(0 == (_dataSize % arrayletLeafSize));
-				GC_SlotObject slotObject(env->getOmrVM(), GC_SlotObject::addToSlotAddress(arrayoidPtr, arrayoidIndex, compressed));
-				slotObject.writeReferenceToSlot(NULL);
-			} else {
-				Assert_MM_true(0 != (_dataSize % arrayletLeafSize));
-				Assert_MM_true(arrayoidIndex == _numberOfArraylets);
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+			if (!indexableObjectModel->isDoubleMappingEnabled())
+#endif /* J9VM_GC_ENABLE_DOUBLE_MAP */
+			{
+				UDATA remainderBytes = _dataSize % arrayletLeafSize;
+				if (0 != remainderBytes) {
+					Assert_MM_true((indexableObjectModel->getSpineSize(spine) + remainderBytes + env->getObjectAlignmentInBytes()) > arrayletLeafSize);
+				}
```

Comment:
I'm on a fence, if we really need this assert. @dmitripivkine, opinion?
Besides, for Metronome that does not move object, even stronger assert (without +objectAlignment) should hold, right? 
And finally, you have a clone of this in another file - no way to reuse?

Reply:
Yes you're right. I'll reuse the one from `ArrayletObjectModel.cpp`. And yes, now that a discontiguous arraylets can have sizes that are not just modulo of `arrayletLeafSize`, I wasn't sure myself if we should have these Asserts or not.